### DISCUSSION TITLE
The README cannot be read when publishing with lerna

### DISCUSSION THREAD

**Discussion by xiangshoulai** [verdaccio/verdaccio#5002] [Q&A]
**Your Environment**
  
 * **verdaccio version**: v6.0.2
 * **node version** [21.1.0]:
 * **package manager**: [npm@10.2.0]
 * **os**: [mac,  linux]
 * **platform**: [npm]

**Describe the bug**
With lerna, this happens, but with npm publish, it doesn't. The problem is described below:

```
ERROR: No README data found!
```

**Comment by xiangshoulai** [comment]
The result is:

![image](https://github.com/user-attachments/assets/67e1b011-dba0-4fb4-8338-abab4ae1aeae)

The downloaded zip contains the README file：

![image](https://github.com/user-attachments/assets/c1d53736-ca6b-4c67-925e-872597a3a2c3)

**Comment by juanpicado** [comment]
Could you share how the Lerna setup looks like in your side ?

**Comment by xiangshoulai** [comment]
![image](https://github.com/user-attachments/assets/daefc919-aca6-4d81-903a-f63747c3cfb1)

**Comment by mbtools** [comment]
I tested it with the [lerna example](https://github.com/lerna/getting-started-example) repo and Verdaccio works fine. 

Lerna does *not* take the readme file in the root folder. You have to add a readme to each package.

https://lerna.js.org/docs/troubleshooting#publishing-to-a-private-npm-registry-artifactory-npm-enterprise-etc

**Comment by xiangshoulai** [comment]
Yes, all packages have README files
![image](https://github.com/user-attachments/assets/8fd3c8b3-acdf-4a59-a322-551808291837)

**Comment by mbtools** [comment]
Which version of lerna? What's in package.json?

**Comment by xiangshoulai** [comment]
> Which version of lerna? What's in package.json?

8.1.9

**Comment by mbtools** [comment]
Did you try adding `.npmrc` as recommended [here](https://lerna.js.org/docs/troubleshooting#publishing-to-a-private-npm-registry-artifactory-npm-enterprise-etc)?

**Comment by xiangshoulai** [comment]
> Did you try adding `.npmrc` as recommended [here](https://lerna.js.org/docs/troubleshooting#publishing-to-a-private-npm-registry-artifactory-npm-enterprise-etc)?
 
yes

![image](https://github.com/user-attachments/assets/3c474094-1a93-4d4e-83f4-2a4a7bb9c484)


![image](https://github.com/user-attachments/assets/3e3486eb-d310-4579-991e-727556c41d77)

But here's the result:

![image](https://github.com/user-attachments/assets/251bb766-3224-4457-ba8c-88250ce0164f)

**Comment by juanpicado** [comment]
**On publish a project, the package manager (which on this case is being used by Lerna) includes the README as text on the request body, this is being used by private registries.** 

I don't see neither with pnpm or npm as package manager at Lerna the `readme` content is being send. If you test with non Lerna project the `readme` content is a property of the body as text, this is being used as readme. 

So I don't think so this is a verdaccio issue, rather Lerna is not picking up the readme content correctly. 

<img width="798" alt="Screenshot 2024-12-15 at 12 38 26" src="https://github.com/user-attachments/assets/51a6c58e-2860-4cc5-ab9a-c08567fe7393" />

this is how it looks like with the `readme` as property. 

<img width="845" alt="Screenshot 2024-12-15 at 12 41 45" src="https://github.com/user-attachments/assets/f93066e9-6494-4cfe-8b78-1ed06506ed0e" />
